src/Features/Core/Portable/ReplaceMethodWithProperty/AbstractReplaceMethodWithPropertyService.cs

```diff
@@ -47,17 +47,26 @@ private static bool OverridesMetadataSymbol(IMethodSymbol method)
         {
             var getMethodDeclaration = getAndSetMethods.GetMethodDeclaration;
             var setMethodDeclaration = getAndSetMethods.SetMethodDeclaration;
-            if (setMethodDeclaration == null)
+            var finalLeadingTrivia = getAndSetMethods.GetMethodDeclaration.GetLeadingTrivia().ToList();
+
+            if (getMethodDeclaration.ChildNodes().Any(n => n.RawKind == syntaxFacts.SyntaxKinds.ParameterList && n.GetTrailingTrivia().Any(t => !syntaxFacts.IsWhitespaceOrEndOfLineTrivia(t))))
             {
-                return property.WithLeadingTrivia(getMethodDeclaration.GetLeadingTrivia());
+                finalLeadingTrivia.AddRange(getMethodDeclaration.ChildNodes().Where(n => n.RawKind == syntaxFacts.SyntaxKinds.ParameterList).First().GetTrailingTrivia());
             }
```

Comment:
this all needs comments.

Reply:
I added comments to the lines.  See if they suffice. If you'd like more detailed (comments on each of the Linq items) I can add those.